### PR TITLE
Check for `allowedBlocks` === true for payments intro

### LINKED ISSUE #28523: [Earn] Fix the Payments block selector
Originally spotted by @n3f:

> The payments block appears that lets you choose:

> But only the middle button works... Anyone else seeing something similar. (It's the same for me on trunk).

I was trying to point out when did this problem occur, and even after rolling back to mid-december, the problem is still there. It seems unrelated to our paid-newsletter work, but nevertheless - seems like a problem

## Reproduction path

This will be only possible on WPCOM testing flow as the premium content block is only available on wpcom:

1. Sync to wpcom
2. insert `/payment` on a site with enabled payment features
3. Observe that donation and premium content blocks are not able to be selected.

### PR DESCRIPTION
**Pull request by n3f**
Fixes #28523 (again).

The previous fix would not take the case where `settings.allowedBlockTypes` could be `true`.

## Proposed changes:

* One line change to check when all blocks are allowed.

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

* Add a `Payments Button` block
* Selecting a block should work
Or see #28523

### CODE REVIEW

**Review by millerf — APPROVED** [review]
LGTM 🚢

File: projects/plugins/jetpack/changelog/earn-fix-payments-intro-blocks-true [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: bugfix
+Comment: one line change -- should be obvious
```
Comment (kraftbj) [inline projects/plugins/jetpack/changelog/earn-fix-payments-intro-blocks-true:3]: Approving, but please do put something here that's useful. The changelogs are used outside of the context of the PR itself so, in and of itself, this changelog is not obvious.

**Review by kraftbj — APPROVED** [review]

**Comment by samiff** [comment]
cherry-picked to `weekly/branch-2023-05-08` in 65eb3ffcbdfcd49bf30499fe63090c2ea1706e5c

File: projects/plugins/jetpack/changelog/earn-fix-payments-intro-blocks-true (hunk 1 above)
Comment (n3f) [reply projects/plugins/jetpack/changelog/earn-fix-payments-intro-blocks-true:3]: Oops, sorry!

**Pull request merged**